engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1717,10 +1717,11 @@ public Class<?> visit(UnaryExpr n, VisitArgs printer) {
         // since the original expression was visited by getTypeWithCaching at the beginning of this method.
         final Class<?> result = unaryOpOverloadMethod.accept(this, printer);
 
-        // Verify that the operator overload method returns the original expected type:
-        Assert.equals(ret, "ret", result, "result");
+        // Verify that the operator overload method returns the original expected type (or its unboxed form):
```

Comment:
We talked about "smart unboxing" when given a `null` input that matches a single primitive method. I'm not sure if it makes sense, yet.